src/textual/app.py

```diff
@@ -202,6 +209,105 @@ def exit(self, result: ReturnType | None = None) -> None:
         self._return_value = result
         self.close_messages_no_wait()
 
+    @property
+    def focus_chain(self) -> list[Widget]:
+        """Get widgets that may receive focus, in focus order."""
+        widgets: list[Widget] = []
+        add_widget = widgets.append
+        root = self.screen
+        stack: list[Iterator[Widget]] = [iter(root.children)]
```

Comment:
shouldn't we use `root.displayed_children` , so children that have `display: none` are not included in the focus chain?

Reply:
Yep